src/shared.c

```diff
@@ -237,25 +237,28 @@ void Vector4Lerp(const vec4_t from, const vec4_t to, const vec_t frac, vec4_t ou
  * @brief Produces the linear interpolation of the two angles for the given fraction.
  * Care is taken to keep the values between -180.0 and 180.0.
  */
-void AngleLerp(const vec3_t from, const vec3_t to, const vec_t frac, vec3_t out) {
-	vec3_t _from, _to;
+vec_t AngleLerp(vec_t from, vec_t to, const vec_t frac) {
 
-	// copy the vectors to safely clamp this lerp
-	VectorCopy(from, _from);
-	VectorCopy(to, _to);
+	if (to - from > 180.0) {
+		to -= 360.0;
+	}
 
-	for (int32_t i = 0; i < 3; i++) {
+	if (to - from < -180.0) {
+		to += 360.0;
+	}
 
-		if (_to[i] - _from[i] > 180.0) {
-			_to[i] -= 360.0;
-		}
+	return Lerp(from, to, frac);
+}
 
-		if (_to[i] - _from[i] < -180.0) {
-			_to[i] += 360.0;
-		}
-	}
```

Comment:
Little scared here. View angles runs through this code path, so let's be sure to test riding rotators, all sorts of teleporters and other things that might trigger a bad clamp.